From c1ba7642b7a8567fd945e04cc9ad9e1bd1014576 Mon Sep 17 00:00:00 2001
From: Chris Bagwell <chris@cnpbagwell.com>
Date: Sun, 4 Sep 2011 22:10:01 -0500
Subject: [PATCH] look at current packet during 2 finger gesture lag

Probably the original logic meant to be doing something like:

"if (ds[0].proximity && dsLast[0].proximity && ds[1].proximity)"

but it left out the ds[0].proximity part.  This had two issues:

1) We got 1 extra packet worth of movement because of delay in
going into LAG mode upon 2 finger touch.
2) It could go into LAG mode upon releasing 1st finger before
second finger.

Signed-off-by: Chris Bagwell <chris@cnpbagwell.com>
---
 src/wcmTouchFilter.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/wcmTouchFilter.c b/src/wcmTouchFilter.c
index a4a64f6..0186799 100644
--- a/src/wcmTouchFilter.c
+++ b/src/wcmTouchFilter.c
@@ -219,7 +219,7 @@ void wcmGestureFilter(WacomDevicePtr priv, int channel)
 	 * prevents cursor movement.  Force to LAG mode if ever in NONE
 	 * mode to stop cursor movement.
 	 */
-	if (dsLast[0].proximity && ds[1].proximity)
+	if (ds[0].proximity && ds[1].proximity)
 	{
 		if (!common->wcmGestureMode)
 			common->wcmGestureMode = GESTURE_LAG_MODE;
-- 
1.7.4.1

